From f972f1bbc794f41aa72f5a4a68f75c115a919da3 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Wed, 26 Mar 2014 19:03:52 +0800
Subject: [PATCH 3819/5965] m6tvd: add vdin config for miracast source function

by wanhu.zheng@amlogic.com

and update defconfig by make savedefconfig.
---
 arch/arm/boot/dts/amlogic/meson6tvd_ref.dtd | 31 ++++++++++++++++++++-
 arch/arm/configs/meson6tvd_defconfig        |  3 +-
 2 files changed, 32 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson6tvd_ref.dtd b/arch/arm/boot/dts/amlogic/meson6tvd_ref.dtd
index 836d7bd794d0..a36d37ddcb33 100755
--- a/arch/arm/boot/dts/amlogic/meson6tvd_ref.dtd
+++ b/arch/arm/boot/dts/amlogic/meson6tvd_ref.dtd
@@ -189,6 +189,33 @@
 		reserve-iomap = "true";
 	};
 
+/// ***************************************************************************************
+///	-	DISP&MM-AVC Video Encoder
+//$$ MODULE = "DISP&MM-AVC Video Encoder"
+//$$ DEVICE="amvenc_avc"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2 ="reg"
+	amvenc_avc {
+		compatible = "amlogic,amvenc_avc";
+		dev_name = "amvenc_avc.0";
+		status = "okay";
+		reserve-memory = <0x00f00000>;
+		reserve-iomap = "true";
+	};
+
+/// ***************************************************************************************
+///	-	DISP&MM-AMLVIDEO2
+//$$ MODULE = "DISP&MM-AMLVIDEO2"
+//$$ DEVICE="amlvideo2"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2 ="reg"
+	amlvideo2 {
+		compatible = "amlogic,amlvideo2";
+		dev_name = "amlvideo2.0";
+		status = "okay";
+		reserve-memory = <0x01000000>;
+		reserve-iomap = "true";
+	};
 
 /// ***************************************************************************************
 ///	-	DISP&MM-Vout
@@ -685,6 +712,7 @@
 		dev_name = "vdin0";
 		status = "ok";
 		reserve-memory = <0x2400000>;
+		camera_size = <2592 1944>;//camera is 5647,the picture resolution is 2592x1944;so 2592x1944x2x4=41M
 		reserve-iomap = "true";
 		irq = <115>;
 		vdin_id = <0>;
@@ -694,7 +722,8 @@
 		compatible = "amlogic,vdin";
 		dev_name = "vdin1";
 		status = "ok";
-		reserve-memory = <0x2400000>;
+		reserve-memory = <0x1000000>;
+		miracast_size = <1920 1080>;//miracast resolution is 1024x768,1024x768x2x4=6M
 		reserve-iomap = "true";
 		irq = <117>;
 		vdin_id = <1>;
diff --git a/arch/arm/configs/meson6tvd_defconfig b/arch/arm/configs/meson6tvd_defconfig
index 3d3a6ec24d32..a9f504c568bb 100755
--- a/arch/arm/configs/meson6tvd_defconfig
+++ b/arch/arm/configs/meson6tvd_defconfig
@@ -168,7 +168,6 @@ CONFIG_FB_OSD1_DEFAULT_WIDTH=1920
 CONFIG_FB_OSD1_DEFAULT_HEIGHT=1080
 CONFIG_FB_OSD1_DEFAULT_WIDTH_VIRTUAL=1920
 CONFIG_FB_OSD1_DEFAULT_HEIGHT_VIRTUAL=2160
-CONFIG_AM_GE2D=y
 CONFIG_AM_LOGO=y
 # CONFIG_AML_HDMI_TX is not set
 CONFIG_TVIN=y
@@ -178,9 +177,11 @@ CONFIG_TVIN_VBI=y
 CONFIG_TVIN_HDMI=y
 # CONFIG_TVIN_BT656 is not set
 # CONFIG_TVIN_CSI is not set
+CONFIG_VIUIN=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE=y
 CONFIG_AMLOGIC_CAPTURE_FRAME_ROTATE=y
 CONFIG_AMLOGIC_VM_DISABLE_VIDEOLAYER=y
+CONFIG_V4L_AMLOGIC_VIDEO2=y
 CONFIG_AMLOGIC_IONVIDEO=y
 CONFIG_DEINTERLACE=y
 CONFIG_D2D3_PROCESS=y
-- 
2.19.0

